### PR TITLE
v2.0 ios

### PR DESCRIPTION
**Pull request by ksyeo1010**

### CODE REVIEW

File: binding/ios/Leopard.swift [hunk 1]
```diff
@@ -23,18 +23,32 @@ public struct LeopardWord {
...
+
     /// Constructor.
     ///
     /// - Parameters:
     ///   - word: Transcribed word.
     ///   - startSec: Start of word in seconds.
     ///   - endSec: End of word in seconds.
     ///   - confidence: Transcription confidence. It is a number within [0, 1].
-    public init(word: String, startSec: Float, endSec: Float, confidence: Float) {
+    ///   - speakerTag: he speaker tag is `-1` if diarization is not enabled during initialization;
```
Comment (laves) [inline binding/ios/Leopard.swift:38]:
```suggestion
    ///   - speakerTag: The speaker tag is `-1` if diarization is not enabled during initialization;
```

File: binding/ios/Leopard.swift [hunk 2]
```diff
@@ -185,15 +227,9 @@ public class Leopard {
...
-        } catch let error as LeopardInvalidArgumentError {
-            let fileExtension = (audioPath as NSString).pathExtension.lowercased()
-            if !Leopard.supportedAudioTypes.contains(fileExtension) {
-                throw LeopardInvalidArgumentError("Audio file with format '\(fileExtension)' is not supported")
-            } else {
-                throw error
-            }
+        if status != PV_STATUS_SUCCESS {
+            let messageStack = try getMessageStack()
+            throw pvStatusToLeopardError(status, "Leopard process failed", messageStack)
```
Comment (laves) [inline binding/ios/Leopard.swift:232]:
```suggestion
            throw pvStatusToLeopardError(status, "Leopard process file failed", messageStack)
```

File: binding/ios/LeopardAppTest/LeopardAppTestUITests/LeopardAppTestUITests.swift [hunk 3]
```diff
@@ -94,88 +106,75 @@ class LeopardAppTestUITests: XCTestCase {
...
-    func validateMetadata(transcript: String, words: [LeopardWord], audioLength: Float) {
-        let normTranscript = transcript.uppercased()
-        for word in words {
-            XCTAssert(normTranscript.contains(word.word.uppercased()))
-            XCTAssert(word.startSec > 0.0)
-            XCTAssert(word.startSec <= word.endSec)
-            XCTAssert(word.endSec < audioLength)
-            XCTAssert(word.confidence >= 0.0 && word.confidence <= 1.0)
+    func validateMetadata(words: [LeopardWord], expectedWords: [LanguageTestWord], enableDiarization: Bool) {
+        for i in 0..<words.count {
```
Comment (laves) [inline binding/ios/LeopardAppTest/LeopardAppTestUITests/LeopardAppTestUITests.swift:110]: assert len of words == expectedWords

**Review by laves — CHANGES_REQUESTED** [review]

File: binding/ios/Leopard.swift [hunk 4]
```diff
@@ -80,8 +106,12 @@ public class Leopard {
                 accessKey,
                 modelPathArg,
                 enableAutomaticPunctuation,
+                enableDiarization,
```
Comment (laves) [inline binding/ios/Leopard.swift:109]: you're missing the actual call to `pv_set_sdk` - double check that this is present in all your SDKs. If it's missing in one, it'll mess up our stats and we'll have to do a rerelease

**Review by laves — CHANGES_REQUESTED** [review]

**Comment by ksyeo1010** [comment]
react native relies on this PR, this PR relies on RN pr.

**Review by laves — APPROVED** [review]

**Pull request merged**